feature-home/src/main/java/com/agb/feature_home/ui/HomeViewModel.kt

```diff
@@ -3,20 +3,65 @@ package com.agb.feature_home.ui
-import com.agb.core.domain.interactor.UserInteractor
-import com.agb.core.domain.model.User
+import com.agb.feature_cards.core.datasource.CardsDataSource
+import com.agb.feature_cards.core.domain.models.Card
+import com.agb.feature_transactions.core.datasource.TransactionsDataSource
+import com.agb.feature_transactions.core.domain.models.Transaction
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
 
-class HomeViewModel(private val userInteractor: UserInteractor) : ViewModel() {
-    private val _userFlow = MutableStateFlow<Result<User>>(Result.Pending)
-    val userFlow: StateFlow<Result<User>> get() = _userFlow
+class HomeViewModel(
+    private val cardsDS: CardsDataSource,
+    private val transactionsDS: TransactionsDataSource,
+) : ViewModel() {
+    private val _itemsFlow = MutableStateFlow<Result<Map<Card, List<Transaction>>>>(Result.Pending)
+    val itemsFlow: StateFlow<Result<Map<Card, List<Transaction>>>> get() = _itemsFlow
 
-    fun getName() {
+    fun getCardsWithTransactions() {
         viewModelScope.launch {
-            val user = userInteractor.getUser()
-            _userFlow.emit(user)
+            launch {
+                val cards = cardsDS.getCards()
+                val transactions = transactionsDS.getTransactions()
+
+                when {
+                    cards is Result.Success && transactions is Result.Success -> {
+                        val grouped = transactions.data.groupBy { it.cardId }
+                        val res = cards.data.map { it to (grouped[it.cardId] ?: listOf()) }.toMap()
+                        _itemsFlow.emit(Result.Success(res))
+                    }
+                    cards is Result.Error -> {
+                        _itemsFlow.emit(cards)
+                    }
+                    transactions is Result.Error -> {
+                        _itemsFlow.emit(transactions)
+                    }
+                }
+            }
         }
     }
 }
+
+// private val _itemsFlow = MutableStateFlow<Result<List<Pair<Card, List<Transaction>>>>>(Result.Pending)
+// val itemsFlow: StateFlow<Result<List<Pair<Card, List<Transaction>>>>> get() = _itemsFlow
+//
+// fun getCards() {
+//    viewModelScope.launch {
+//        val cards = cardsDS.getCards()
+//        val transactions = transactionsDS.getTransactions()
+//
+//        when {
+//            cards is Result.Success && transactions is Result.Success -> {
+//                val grouped = transactions.data.groupBy { it.cardId }
+//                val res = cards.data.map { it to (grouped[it.cardId] ?: emptyList()) }
+//                _itemsFlow.emit(Result.Success(res))
+//            }
+//            cards is Result.Error -> {
+//                _itemsFlow.emit(cards)
+//            }
+//            transactions is Result.Error -> {
+//                _itemsFlow.emit(transactions)
+//            }
+//        }
+//    }
+// }
```

Comment:
Dead code

Reply:
not dead, it just waiting it's time
![Yoda_SWSB](https://user-images.githubusercontent.com/43153650/120751918-77cd6f00-c511-11eb-9dba-685872cfb7f8.png)